lib/common/memory.c

```diff
@@ -604,7 +604,7 @@ void h2o_append_to_null_terminated_list(void ***list, void *element)
 
 char *h2o_strerror_r(int err, char *buf, size_t len)
 {
-#if !(defined(_GNU_SOURCE) && defined(__gnu_linux__))
+#if (!(defined(_GNU_SOURCE) && defined(__gnu_linux__))) || defined(_MUSL_LIB)
```

Comment:
I do not think this is necessary as IIUC `defined(__gnu_linux__)` evaluates true only if glibc is in use.